drivers/hmis_csv_importer/app/jobs/hmis_csv_importer/cleanup/expire_base_job.rb

```diff
@@ -81,11 +87,64 @@ def process_model(model)
       start_time = Time.current
       # TODO: this can be removed (or at least the count query could be removed once we're comfortable with the results)
       log "Start Processing: #{model.table_name}, rows overall: #{model.with_deleted.count}"
-      model.connection.execute(mark_expired_query(model))
+
+      expire_data(model)
+
       # TODO: this can be removed (or at least the count query could be removed once we're comfortable with the results)
       log "Completed Processing: #{model.table_name}, rows expired: #{model.with_deleted.where(expired: true).count} in #{elapsed_time(Time.current - start_time)}"
     end
 
+    private def expire_data(model)
+      tmp_table_prefix = model.table_name.downcase
+      tmp_table_name = "#{tmp_table_prefix}_tmp_exp"
+      tmp_model_name = "#{model.class.name.demodulize}Expire"
+      tmp_class = HmisCsvImporter::TempTable.create_temporary_table(table_name: tmp_table_name, model_name: tmp_model_name)
+      begin
+        populate_tmp_table(model, tmp_table_name)
+        write_expirations(model, tmp_table_name)
+      ensure
+        tmp_class.drop
```

Comment:
temporary tables are automatically dropped at the end of the session. Probably don't need this?